lib/project.ts

```diff
@@ -488,6 +491,62 @@ export class Project implements Project.IProject {
 		}).future<void>()();
 	}
 
+	private hasConfigurationSpecificDataForProperty(normalizedPropertyName: string): boolean {
+		var properties = _(this.configurationSpecificData)
+			.values()
+			.map(val => _.keys(val))
+			.flatten()
+			.value();
+		return _.some(properties, prop => prop === normalizedPropertyName);
+	}
+
+	private getConfigurationSpecificDataForProperty(normalizedPropertyName: string): any[] {
+		var configsDataForProperty: any[] = [];
+		var numberOfConfigs = _.keys(this.configurationSpecificData).length;
+
+		_.forEach(_.keys(this.configurationSpecificData), config => {
+			var data = this.configurationSpecificData[config][normalizedPropertyName];
+			configsDataForProperty.push(_.flatten([data]));
+		});
+
+		var sharedValues: string[] = _.intersection.apply(null, configsDataForProperty);
+		var valuesInAllConfigs = _.map(sharedValues, value => helpers.fill(value, numberOfConfigs));
+		var configSpecificValues = _.map(configsDataForProperty, config => _.difference(config, sharedValues));
+
+		var maxLength = _(configSpecificValues)
+			.values()
+			.map(value => value.length)
+			.max()
+			.value();
```

Comment:
max() is not chainable, therefore .value() should not be needed. Of course, I did not test it :)